### PR TITLE
Merge logged properties into userInfo dict for Firebase Logger

### PR DESCRIPTION
**Pull request by stevelind-okta**
Propagate properties dict from `log.warning()` and `log.error()` into firebase destination.
Added Unit test for `userInfo` dict creation
Fix multithreading issue with default properties

### CODE REVIEW

File: OktaLogger/OktaLoggerDestination.swift [hunk 1]
```diff
@@ -155,13 +155,15 @@ open class OktaLoggerDestinationBase: NSObject, OktaLoggerDestinationProtocol {
...
+        updatedProperties.merge(defaultProperties, uniquingKeysWith: { (_, last) in last })
+        self.defaultProperties = updatedProperties
     }
 
     open func removeDefaultProperties(for key: AnyHashable) {
-        defaultProperties.removeValue(forKey: key)
-        self.defaultPropertiesDescription = Self.description(of: self._defaultProperties)
+        var updatedProperties = defaultProperties
+        updatedProperties.removeValue(forKey: key)
+        defaultProperties = updatedProperties
```
Comment (stevelind-okta) [inline OktaLogger/OktaLoggerDestination.swift:166]: I was getting a crash on the description line. We only execute the lock on get/set and accessing the `self._defaultProperties` directly is unsafe. 

Changed to use the getter/setter locks which also update the description. Verified in debugging that `updatedProperties` object is a copy not a reference to `self.defaultProperties`

File: OktaLoggerTests/OktaLoggerFileLoggerMultithreadingTests.swift [hunk 2]
```diff
@@ -70,7 +70,13 @@ class OktaLoggerFileLoggerMultithreadingTests: XCTestCase {
         }
 
         wait(for: [testFinishExpectation], timeout: defaultTimeout)
+        
+        let purgeExpectation = expectation(description: "Purge logs from main thread")
         lumberjackDelegate.purgeLogs()
+        DispatchQueue.main.async {
+            purgeExpectation.fulfill()
+        }
+        wait(for: [purgeExpectation], timeout: 1.0)
```
Comment (stevelind-okta) [inline OktaLoggerTests/OktaLoggerFileLoggerMultithreadingTests.swift:79]: was timing out inconsistently, waiting for clear main queue does the job

File: OktaLogger/OktaLoggerDestination.swift (hunk 1 above)
Comment (IldarAbdullin-okta) [inline OktaLogger/OktaLoggerDestination.swift:166]: Ok, so basically `defaultProperties` is like objc `atomic` property. Makes sense 👍

**Review by IldarAbdullin-okta — APPROVED** [review]
LGTM!

**Review by BorysKasianenko-okta — APPROVED** [review]

**Comment by BorysKasianenko-okta** [comment]
Please update `version` in `OktaLogger.podspec` before merge.

**Review by AntonVoitsekhivskyi-okta — APPROVED** [review]
LGTM

**Comment by stevelind-okta** [comment]
> Please update `version` in `OktaLogger.podspec` before merge.

Thanks, good catch!

**Pull request merged**